### PR TITLE
Build: add codeql

### PR DESCRIPTION
**Pull request by aladdin-add**
refs: https://securitylab.github.com/tools/codeql

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ x] Other, please explain:

#### What changes did you make? (Give an overview)

CodeQL: the libraries and queries that power security researchers around the world, as well as code scanning in GitHub Advanced Security (code scanning), LGTM.com, and LGTM Enterprise.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master, 1.x, 2.x, 3.x, 4.x ]
```
Comment (aladdin-add) [inline .github/workflows/codeql-analysis.yml:16]: it is the default value, do we want to remove the 1.x, 2.x, 3.x, 4.x, just keep the master?

**Comment by nzakas** [comment]
There’s not enough info in the pull request to understand why we should add CodeQL or what exactly this PR enables. Can you please explain how this benefits the project and how it actually works?

**Comment by aladdin-add** [comment]
This action runs GitHub's industry-leading semantic code analysis engine, CodeQL, against a repository's source code to find security vulnerabilities. It then automatically uploads the results to GitHub so they can be displayed in the repository's security tab. CodeQL runs an extensible set of queries, which have been developed by the community and the GitHub Security Lab to find common vulnerabilities in your code.

more info can be found in https://github.com/github/codeql-action :)

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (nzakas) [reply .github/workflows/codeql-analysis.yml:16]: Yes. We don’t maintain old versions.

**Review by nzakas — APPROVED** [review]
LGTM

**Pull request merged**